src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -8628,10 +8628,23 @@ def _update_pipeline_run_status(
                 ExecutionStatus.COMPLETED,
                 ExecutionStatus.FAILED,
             }:
-                run_update.end_time = datetime.utcnow()
+                run_update.end_time = datetime.now(timezone.utc)
                 if pipeline_run.start_time and isinstance(
                     pipeline_run.start_time, datetime
                 ):
+                    # We need to ensure both datetimes are timezone-aware to avoid TypeError when subtracting
+                    # Now calculate the duration time
+                    if pipeline_run.start_time.tzinfo is None:
+                        pipeline_run.start_time = (
+                            pipeline_run.start_time.replace(
+                                tzinfo=timezone.utc
+                            )
+                        )
+                    if run_update.end_time.tzinfo is None:
```

Comment:
Can this really happen, we just set the `run_update.end_time` a few lines above for a specific timezone?

Reply:
No, you're right. That can't happen. I hadn't seen that the `run_update_end_time` was set at this point.
I've adapted it.